### ISSUE TITLE
Some feature requests

### ISSUE THREAD

**Issue by 2fersen** [TrimVis/butlarr#10]
Hey there,
I was doing some thinking and came up with some features that would be cool to see in your project!

1. First of all, maybe adding the /series and /movie queue command to the help message would be good.

2. after adding a media and the download started successfully, the queue command should be printed.  But if no download starts, maybe send a message like, media file currently not found, but download will start as soon as its avaliable.

3. Multiple commands for adding media, like /s and /series for series.

4. For series, it would also be pretty cool to be able to start downloading individual episodes or just a season? There are big series out there, and if every user I give access to this bot could potentially fill my entire space pretty fast! :D

5. You already wrote in the to-do list to add more privileges, maybe you already think about giving like the normal user just the right download quality below 4k. Would be a cool thing too, because a lot of people dont know how big 4k files are.

**Comment by TrimVis** [comment]
Thanks a lot for the thoughts! :)

> 1. First of all, maybe adding the /series and /movie queue command to the help message would be good.
Good point!

Currently the commands in general need to be added by hand, that's also why I forgot to add it to the help page :)
I will make sure to add it asap and change how the help menu works, so this doesn't happen again

> 2. after adding a media and the download started successfully, the queue command should be printed. But if no download starts, maybe send a message like, media file currently not found, but download will start as soon as its avaliable.

I like the idea, I feel like in general more information than just "Series added" should be given after something has been added :thinking: 
I'm not sure  it really makes sense to do it just like you said (from a technical standpoint) as there will always be a delay between download start and search, therefore I would need to wait until I send the last notification
However I was thinking of integrating notification subscriptions over time, so by having these we could have a notification for "Download started for ...", which would feel quite similar to what you have described from a user perspective :)

> 3. Multiple commands for adding media, like /s and /series for series.

This is already supported :wink: Might be missing in the setup helper, but its configurable in the config.yml file (simply add another entry to the commands list or array.
```yaml
services:
  - type: "Radarr"
    commands: ["movie", "m"]
    api: "movie"
```

> 4. For series, it would also be pretty cool to be able to start downloading individual episodes or just a season? There are big series out there, and if every user I give access to this bot could potentially fill my entire space pretty fast! :D

Good point! This is definitely necessary! I will look into it :)

> 5. You already wrote in the to-do list to add more privileges, maybe you already think about giving like the normal user just the right download quality below 4k. Would be a cool thing too, because a lot of people dont know how big 4k files are.

Maybe the whole permission system has to be rethought :thinking:
I could however add options to the config to manage what quality profile ids users or mods are allowed to assign :)
Should be rather quick to add and shouldn't break the current role system.

**Comment by TrimVis** [comment]
- [x] Fix help menu (show queue command)
- [x] Add generic help menu logic
- [x] Season (possibly episode selection) when downloading of series
- [ ] Add permissioned quality profiles

**Comment by arunsinghbartwal** [comment]
as we are adding more commands hence we have to take care that the search patterns works with /movie "string" but as we are about to add commands like /movie list or /movie xyz hence it is better to use /movie search "string" instead of using /movie "string"
It will help in listing movies with matching names as command

**Comment by 2fersen** [comment]
About the message after adding a media, maybe the solution like printing "Series added" and after a confirmation from the search were maid, the bot will send another message. I think thats the easiest way to solve it.

Oh, I did not know about the multi-command method, thanks for letting me know! Also I think printing all possible variants of the command to add in the help message would also be great!

I would love to help you code this, but your level of coding is much cleaner and better then mine, I am just a script kiddie :(

**Comment by TrimVis** [comment]
> as we are adding more commands hence we have to take care that the search patterns works with /movie "string" but as we are about to add commands like /movie list or /movie xyz hence it is better to use /movie search "string" instead of using /movie "string" It will help in listing movies with matching names as command

I added a additional search subcommand :)
I will likely update the docker image tomorrow :)

**Comment by Klubas** [comment]
> after adding a media and the download started successfully, the queue command should be printed. But if no download starts, maybe send a message like, media file currently not found, but download will start as soon as its avaliable.

I disagree with this, you can configure radarr/sonarr to send a telegram message everytime it grabs and downloads something. As the search and grab is assyncronous, I feel that is a better solution than implementing that logic in butlarr. 

Unless you are talking about just printing a more "human" message, I don't really see the point

**Comment by TrimVis** [comment]
I will close this issue, as all the features I am planning on including from your post exist in other more specific issues.
As this will likely lead to better discussions (if there are any), I will shift the focus there.

Thanks for all the ideas :)
